src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActions/CodeAnalysisSuggestionsSuggestedActionCallback.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using Microsoft.CodeAnalysis.CodeAnalysisSuggestions;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Options;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.Suggestions;
+
+[Export(typeof(ISuggestedActionCallback))]
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal class CodeAnalysisSuggestionsSuggestedActionCallback(IThreadingContext threadingContext, IGlobalOptionService globalOptions)
+    : ForegroundThreadAffinitizedObject(threadingContext), ISuggestedActionCallback
+{
+    private static readonly ImmutableHashSet<string> s_codeQualityAnalyzerAndFixerAssemblyNames = ImmutableHashSet.Create(
+        StringComparer.OrdinalIgnoreCase,
+        "Microsoft.CodeAnalysis.NetAnalyzers",
+        "Microsoft.CodeAnalysis.CSharp.NetAnalyzers",
+        "Microsoft.CodeAnalysis.VisualBasic.NetAnalyzers");
+
+    private static readonly ImmutableHashSet<string> s_codeStyleAnalyzerAndFixerAssemblyNames = ImmutableHashSet.Create(
+        StringComparer.OrdinalIgnoreCase,
+        "Microsoft.CodeAnalysis.Features",
+        "Microsoft.CodeAnalysis.CSharp.Features",
+        "Microsoft.CodeAnalysis.VisualBasic.Features",
+        "Microsoft.CodeAnalysis.CodeStyle",
+        "Microsoft.CodeAnalysis.CSharp.CodeStyle",
+        "Microsoft.CodeAnalysis.VisualBasic.CodeStyle",
+        "Microsoft.CodeAnalysis.CodeStyle.Fixes",
+        "Microsoft.CodeAnalysis.CSharp.CodeStyle.Fixes",
+        "Microsoft.CodeAnalysis.VisualBasic.CodeStyle.Fixes");
+
+    private readonly IGlobalOptionService _globalOptions = globalOptions;
+
+    public void OnSuggestedActionExecuted(SuggestedAction action)
+    {
+        // We'll need to be on the UI thread for the operations.
+        AssertIsForeground();
+
+        // If the user has disabled the feature, then we bail out immediately.
+        if (!_globalOptions.GetOption(CodeAnalysisSuggestionsOptionsStorage.ShowCodeAnalysisSuggestionsInLightbulb))
+        {
+            return;
+        }
+
+        // Update option values for first party code quality and code style fixes.
+        var assemblyName = action.Provider.GetType().Assembly.GetName().Name;
+        UpdateOptions(_globalOptions, assemblyName, codeQuality: true);
+        UpdateOptions(_globalOptions, assemblyName, codeQuality: false);
+        return;
+
+        static void UpdateOptions(IGlobalOptionService options, string assemblyName, bool codeQuality)
+        {
+            var isFirstPartyFix = codeQuality
```

Comment:
if i understand the code correctly, it's codeQuality if containing assembly is in s_codeQualityAnalyzerAndFixerAssemblyNames, or codeStyle if it's in s_codeStyleAnalyzerAndFixerAssemblyNames?

if so, could we do something like this instead of calling the method twice?

```cs
if (s_codeQualityAnalyzerAndFixerAssemblyNames.Contains(assemblyName))
{
    codeQuality = true;
}
else if (s_codeStyleAnalyzerAndFixerAssemblyNames.Contains(assemblyName))
{ 
    codeQuality = false;
}
else
{
    return;
}
```